File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        counts = defaultdict(int)

        for n in nums:
            counts[n] += 1
        
        buckets = [[] for _ in range(len(nums))]
        for n in counts:
            buckets[counts[n] - 1].append(n)
        
        results = []
        for i in range(len(nums), 0, -1):
            results.extend(buckets[i - 1])
        
        return results[:k]

File: Data Structures & Algorithms/is-anagram/submission-3.go
func isAnagram(s string, t string) bool {
    countS := [26]int{}
    countT := [26]int{}
    for _, c := range s {
        countS[c - 'a'] += 1
    }
    for _, c := range t {
        countT[c - 'a'] += 1
    }
    return countS == countT
}
